objstore/objstore.go

```diff
@@ -0,0 +1,112 @@
+package objstore
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"fx.prodigy9.co/config"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+type StorageClient struct {
+	accessKey string
+	secretKey string
+	url       string
+	bucket    string
+	client    *minio.Client
+}
+
+var (
+	StorageAccessKeyConfig = config.Str("STORAGE_ACCESS_KEY")
+	StorageSecretKeyConfig = config.Str("STORAGE_SECRET_KEY")
+	StorageURLConfig       = config.Str("STORAGE_URL")
+	StorageBucketConfig    = config.Str("STORAGE_BUCKET_NAME")
+
+	DefaultClient *StorageClient
+)
+
+func NewClient(cfg *config.Source) *StorageClient {
+	return newS3Client(cfg)
+}
+
+func getDefaultClient() {
```

Comment:
It's a bit weird that `DefaultClient` variable is public, but this getter (which will build the client if its nil) is private.

Perhaps, better to make client "zero-value usable" instead?

Reply:
removed. by changing to calling tryGetClient and then checking if there is a client or not, if not, will do getClient